botocore/paginate.py

```diff
@@ -323,22 +325,35 @@ def _parse_starting_token(self):
             except ValueError:
                 raise ValueError("Bad starting token: %s" %
                                  self._starting_token)
-        for part in parts:
+        for token_index, part in enumerate(parts):
```

Comment:
So I was not sure. After looking at the code, it is defined by the order in which it is defined in the JSON paginator model. So if someone was to add a new input/ouput token that broke the order in the list that will break this logic, but it will also break token parsing logic elsewhere... So it should be fine to rely on ordering I suppose.